tests/models/test_taskinstance.py

```diff
@@ -1475,6 +1476,8 @@ def test_handle_failure(self):
                               dag=dag)
         ti1 = TI(task=task1, execution_date=start_date)
         ti1.state = State.FAILED
+        ti1.start_date = start_date
+        ti1.end_date = end_date
```

Comment:
Why was this change needed? The tests should pass without this change, no?

Reply:
the original structure of the method under test had a wide scope for the try/except block that was intended to log when we're unable to send an email and move on. i think TypeError's due to these attributes being none were being caught by that block and allowing the test to complete without error

Reply:
for more context, here is the failure that cropped up during the execution of existing test code after I narrowed the scope of the try/except to only be around sending the email. my mistake, it was an AttributeError not a TypeError :)

```
        self.log.info(
            '%s dag_id=%s, task_id=%s, execution_date=%s, start_date=%s, end_date=%s',
            log_message,
            self.dag_id,
            self.task_id,
            self.execution_date.strftime('%Y%m%dT%H%M%S'),
>           self.start_date.strftime('%Y%m%dT%H%M%S'),
            self.end_date.strftime('%Y%m%dT%H%M%S')
        )
E       AttributeError: 'NoneType' object has no attribute 'strftime'
```

Reply:
Aha! https://github.com/apache/airflow/pull/7257

Reply:
With that PR is this change now not needed? (Does it make sense to keep it anyway?)

Reply:
Actually, I needed to do something to make sure my boiling down of the logging.info call into one spot also incorporated that safety check. I've done that now and removed the start_date/end_date assignments from the test again